sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java

```diff
@@ -922,6 +922,11 @@ private void setupPubSubResources(long now) throws IOException {
     switch (configuration.sourceType) {
       case DIRECT:
         source = sourceEventsFromSynthetic(p);
+        if (configuration.generateInputFileOnly) {
+          PCollection<Event> events = source;
+          source = null;
```

Comment:
is this needed?

Reply:
yeah, if the source is not set to null then the actual query body will get executed

Reply:
this maybe the wrong place to put the transform, can't we swap out the steps after the source?

Reply:
sorry that I might have came out wrong, the logic later down the line is that if the source is null, following steps of pipeline will not get added.
see [this part](https://github.com/apache/beam/blob/009677318246ff5f0ff8370d8f3bf4ca40322343/sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/NexmarkLauncher.java#L1176)

Reply:
I see, sounds good.